### PR TITLE
rabbitmq-server: Fix reported version

### PR DESCRIPTION
**Pull request by dasJ**
Fixes the version in `rabbitmqctl --version` and the web interface

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by xokdvium** [comment]
Would it be possible to add a version test in `passthru.tests` or via `versionCheckHook`?

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367395`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
  </ul>
</details>

**Comment by lucasewd** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367395`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>rabbitmq-server</li>
    <li>rabbitmq-server.doc</li>
    <li>rabbitmq-server.man</li>
  </ul>
</details>

**Comment by lucasew** [comment]
> Would it be possible to add a version test in `passthru.tests` or via `versionCheckHook`?

I'd advocate for the hook because it runs in checkPhase instead of passthru.tests and nixpkgs-review today doesn't look into passthru.tests. That check would always run in a bump.

**Comment by dasJ** [comment]
@xokdvium done

@lucasew I'd advocate for less bot spam in my inbox, but at least one of us gets their wish in form of a `installCheckPhase`.

**Comment by lucasew** [comment]
> @xokdvium done
> 
> @lucasew I'd advocate for less bot spam in my inbox, but at least one of us gets their wish in form of a `installCheckPhase`.

I did the bot yesterday. I still don't know exactly what kind of approach would be better. Suggestions welcome.

File: pkgs/by-name/ra/rabbitmq-server/package.nix [hunk 1]
```diff
@@ -106,6 +107,19 @@ stdenv.mkDerivation rec {
     rm $out/INSTALL
   '';
 
+  # Can not use versionCheckHook since that doesn't allow for setting environment variables
+  # which is necessary since Erlang needs a $HOME for the Cookie.
```
Comment (xokdvium) [inline pkgs/by-name/ra/rabbitmq-server/package.nix:111]: `testers.testVersion` can be used with custom env variables. Something along the lines of: https://github.com/NixOS/nixpkgs/blob/b10e42649fd8fd7aff02e83c7853c80eefe92644/pkgs/by-name/fe/feather/package.nix#L75-L80

Comment (xokdvium) [reply pkgs/by-name/ra/rabbitmq-server/package.nix:111]: > I'd advocate for the hook

Fine by me, but I'd like for someone else to chime in as well.

**Review by xokdvium — APPROVED** [review]

**Review by samueltardieu — APPROVED** [review]

**Pull request merged**